client/consensus/src/collators/lookahead.rs

```diff
@@ -98,8 +356,26 @@ pub struct Params<BI, CIDP, Client, Backend, RClient, CHP, SO, Proposer, CS, GOH
 }
 
 /// Run async-backing-friendly for Tanssi Aura.
-pub fn run<Block, P, BI, CIDP, Client, Backend, RClient, CHP, SO, Proposer, CS, GOH>(
-    mut params: Params<BI, CIDP, Client, Backend, RClient, CHP, SO, Proposer, CS, GOH>,
+pub fn run<
+    GSD,
+    Block,
+    P,
+    BI,
+    CIDP,
+    Client,
+    Backend,
+    RClient,
+    CHP,
+    SO,
+    Proposer,
+    CS,
+    GOH,
+    OClient,
+    OBlock,
+>(
+    mut params: Params<GSD, BI, CIDP, Client, Backend, RClient, CHP, SO, Proposer, CS, GOH>,
+    orchestrator_tx_pool: Arc<FullPool<OBlock, OClient>>,
+    orchestrator_client: Arc<OClient>,
```

Comment:
This function should only have 1 param, `params`, so move these two new params inside that struct

Reply:
I put those two fields outside `params` as it is too large with too many generics. 

Is there any specific reason why you would prefer to put those two new argument inside `params` struct?

Reply:
Yes, so that we can call using the syntax `run(Params { orchestrator_tx_pool, orchestrator_client })`, in a way that the order of the arguments doesn't matter. The reason `Params` structs exists is because the `run` function had too many arguments in the first place.

Reply:
Make sense. Will make changes.